magnum_cluster_api/cmd/image_loader.py

```diff
@@ -79,18 +84,24 @@ def main(repository, parallel, insecure):
              https://github.com/google/go-containerregistry/blob/main/cmd/crane/README.md#installation"""
         )
 
-    # NOTE(mnaser): This list must be maintained manually because the image
-    #               registry must be able to support a few different versions
-    #               of Kubernetes since it is possible to have multiple
-    #               clusters running different versions of Kubernetes at the
-    #               same time.
-    images = set(
-        _get_all_kubeadm_images()
-        + _get_calico_images()
-        + _get_cilium_images()
-        + _get_cloud_provider_images()
-        + _get_infra_images()
-    )
+    if manifest:
+        with open(manifest, 'r') as file:
+            manifest_data = yaml.safe_load(file)
+
+        images = set(manifest_data['images'])
+
+    else:
+        # NOTE(mnaser): This list must be maintained manually because the image
+        #               registry must be able to support a few different versions
+        #               of Kubernetes since it is possible to have multiple
+        #               clusters running different versions of Kubernetes at the
+        #               same time.
+        images = set(
+            _get_all_kubeadm_images()
+            + _get_calico_images()
+            + _get_cloud_provider_images()
+            + _get_infra_images()
+        )
```

Comment:
instead of keeping this code, could we supply a default manifest that's equivalent so people don't have to look into two different places (code and manifest) to figure this out?

Reply:
That would extend the scope of this PR as we would have to keep the default manifest in the source tree, manage its installation to a known location and then load it by default. I am conscious that we are making very little progress toward getting this merged so would like to keep the scope contained.

An extracted default manifest would be missing the images found through running kubeadm so I think that this might need more consideration, and should probably be a follow-up PR if anyone were wanting to do that work.